### PR TITLE
chore(): allow the CLI tool itself to use TypeScript.

### PR DESCRIPTION
**Pull request by hansl**
This does not work for the Broccoli (yet).

### CODE REVIEW

File: addon/ng2/commands/new.ts [hunk 1]
```diff
@@ -23,35 +19,35 @@ module.exports = Command.extend({
...
+    run: function(commandOptions, rawArgs) {
+        const packageName = rawArgs.shift();
 
         if (!packageName) {
-            message = chalk.yellow('The `ng ' + this.name + '` command requires a ' +
-            'name argument to be specified. For more details, use `ng help`.');
-
-            return Promise.reject(new SilentError(message));
+            return Promise.reject(new SilentError(
+              `The "ng ${this.name}" command requires a name argument to be specified. ` +
```
Comment (filipesilva) [inline addon/ng2/commands/new.ts]: You dropped the use of `chalk.yellow` here, was it intended?

File: index.js [hunk 2]
```diff
@@ -1,8 +1,6 @@
 /* jshint node: true */
```
Comment (filipesilva) [inline index.js]: Is this file used? `Package.json` points at `lib/cli/index.js` as the main file, so I'm not sure if this one is used by anything.

File: lib/cli/index.js [hunk 3]
```diff
@@ -1,27 +1,61 @@
...
+const cli = require('ember-cli/lib/cli');
+const path = require('path');
+
+
+// This file hooks up on require calls to transpile TypeScript.
+const fs = require('fs');
+const ts = require('typescript');
+const old = require.extensions['.ts'];
+
+require.extensions['.ts'] = function(m, filename) {
```
Comment (filipesilva) [inline lib/cli/index.js]: Is this the same as using `ts-node`? Is there any difference?

**Comment by filipesilva** [comment]
Oh man this is nice! What's stopping you from just bringing in the brocolli `ts` files though?

File: addon/ng2/commands/new.ts (hunk 1 above)
Comment (hansl) [inline addon/ng2/commands/new.ts]: It's an error, it's weird to use a yellow() text.

File: lib/cli/index.js (hunk 3 above)
Comment (hansl) [inline lib/cli/index.js]: `ts-node` was including too many definition files and it was showing duplicates.

**Comment by hansl** [comment]
The broccoli should only be `js` files because these are `require()`d by apps. We could use `ts` files for broccoli but we'll need publish the javascript files to NPM. For now I want to avoid making the publishing process more complex.

File: index.js (hunk 2 above)
Comment (hansl) [inline index.js]: This file is loaded by the `ng` script, and when you `require('angular-cli')` since it's the `"main"` file in package.

Comment (hansl) [inline index.js]: Oh that file, not the other one.  Mmmmh, not sure. I'll leave it here and ask Rob.

File: addon/ng2/commands/new.ts [hunk 4]
```diff
@@ -1,19 +1,15 @@
-'use strict';
```
Comment (jkuri) [inline addon/ng2/commands/new.ts]: I believe you should add the typings at the first line;
`/// <reference path="../../../typings/main.d.ts" />`
also, run `typings install` in `package.json` in postinstall script.
If not, just ignore this.

**Comment by jkuri** [comment]
Good work. LGTM.

**Pull request merged**